### PR TITLE
Add observer prime method

### LINKED ISSUE #120: Don't start with a Kalman gain of zero
It takes some time for the state estimate to converge as well as the Kalman gain. We don't want to be sending dumb torque values so use a better initial condition.

### PR DESCRIPTION
**Pull request by oliverlee**
Fixes #120

### CODE REVIEW

File: projects/flimnap/main.cc [hunk 1]
```diff
@@ -112,22 +112,27 @@ namespace {
     template <typename T>
     struct observer_initializer{
         template <typename S = T>
-        typename std::enable_if<std::is_same<S, observer::Kalman<model_t>>::value, void>::type
-        initialize(S& observer) {
+        typename std::enable_if<std::is_same<typename S::observer_t, observer::Kalman<model_t>>::value, void>::type
+        initialize(S& bicycle) {
+            S::observer_t& observer = bicycle.observer();
```
Comment (oliverlee) [inline projects/flimnap/main.cc]: missing `typename`

File: projects/flimnap/main.cc [hunk 2]
```diff
@@ -112,22 +112,27 @@ namespace {
     template <typename T>
     struct observer_initializer{
         template <typename S = T>
-        typename std::enable_if<std::is_same<S, observer::Kalman<model_t>>::value, void>::type
-        initialize(S& observer) {
+        typename std::enable_if<std::is_same<typename S::observer_t, observer::Kalman<model_t>>::value, void>::type
+        initialize(S& bicycle) {
+            typename S::observer_t& observer = bicycle.observer();
+            observer.set_Q(parameters::defaultvalue::kalman::Q(observer.dt()));
+            /* Reduce steer measurement noise covariance */
```
Comment (mickvangelderen) [inline projects/flimnap/main.cc]: Prefer `//` for comments about blocks of code `/* */` for documenting functions/variable. I suspect the comment also needs to be reflowed/re-wrapped.

File: projects/flimnap/main.cc [hunk 3]
```diff
@@ -112,22 +112,27 @@ namespace {
...
+            observer.set_R(parameters::defaultvalue::kalman::R/1000);
+
+            /* prime the Kalman gain matrix */
```
Comment (mickvangelderen) [inline projects/flimnap/main.cc]: Prefer `//` for comments about blocks of code `/* */` for documenting functions/variable. I suspect the comment also needs to be reflowed/re-wrapped.

File: projects/flimnap/main.cc [hunk 4]
```diff
@@ -112,22 +112,27 @@ namespace {
...
+            bicycle.prime_observer();
+
             /* We start with steer angle equal to the measurement and all other state elements at zero.  */
```
Comment (mickvangelderen) [inline projects/flimnap/main.cc]: Prefer `//` for comments about blocks of code `/* */` for documenting functions/variable. I suspect the comment also needs to be reflowed/re-wrapped.

File: projects/src/simbicycle.hh [hunk 5]
```diff
@@ -161,6 +162,21 @@ void Bicycle<T, U, V>::update_kinematics() {
     m_pose.steer = steer;
 }
 
+template <typename T, typename U, typename V>
```
Comment (mickvangelderen) [inline projects/src/simbicycle.hh]: If T, U and V are of a specific .. category, for lack of a better word, then maybe it makes sense to name them less abstractly?

File: projects/flimnap/main.cc (hunk 2 above)
Comment (oliverlee) [inline projects/flimnap/main.cc]: I've been in consistent with comment type so thanks for bringing this up and I'll make your suggested change.

File: projects/src/simbicycle.hh (hunk 5 above)
Comment (oliverlee) [inline projects/src/simbicycle.hh]: 👍

Comment (oliverlee) [inline projects/src/simbicycle.hh]: I would prefer to do this after merging all the pending PRs as there are a number of dependencies.

**Pull request merged**